### PR TITLE
Add optional dimples on tabs.

### PR DESCRIPTION
**Pull request by jimmc**

### CODE REVIEW

**Comment by jimmc** [comment]
Here's a screenshot showing the two new fields in the form (dimple height and width) and a view of some box pieces with dimples.
screen shot 2017-04-20 at 11 12 43 am

**Comment by paulh-rnd** [comment]
Thanks @jimmc for your contributions and I'm so sorry for taking nearly 3 years (!) to get to this. This looks really useful. I'm going to look at it today with the other PR's and I'll merge this in.

**Comment by mwinoto** [comment]
@paulh-rnd 
I haven't looked at all the other pull requests yet, so maybe some one else has done this. But could you provide a description of the steps that are taken to generate the drawings from the pieces list?

I started modifying this because I needed a box generator for a project, and I still have some need for it so I'd like to clean up the code a bit more.

**Comment by paulh-rnd** [comment]
@mwinoto I'm not sure I understand, are you just wanting the code to be better commented (very reasonable request), or do you need something more?

**Comment by mwinoto** [comment]
@paulh-rnd Yeah, better commented. Calls to drawS() can be hard to follow. It's not obvious how the dividers are generated since that code is inside the loop for pieces. Just a description of what the loop attempts to do so that I can refactor and clean it up.
